### PR TITLE
Add 3D NDT sensor model and tests.

### PR DESCRIPTION
**Pull request by serraramiro1**
### Proposed changes

Small tweaks to enable NDT sensor model for SE3 states and 3D lidar returns.

#### Type of change

- [ ] 🐛 Bugfix (change which fixes an issue)
- [x] 🚀 Feature (change which adds functionality)
- [ ] 📚 Documentation (change which fixes or extends documentation)

### Checklist

_Put an `x` in the boxes that apply. This is simply a reminder of what we will require before merging your code._

- [x] Lint and unit tests (if any) pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] All commits have been signed for [DCO](https://developercertificate.org/)

### CODE REVIEW

File: beluga/include/beluga/sensor/ndt_sensor_model.hpp [hunk 1]
```diff
@@ -105,17 +105,28 @@ inline std::vector<NDTCell<NDim, Scalar>> to_cells(
...
+    Eigen::Vector2i{0, 1},    //
+    Eigen::Vector2i{1, -1},   //
+    Eigen::Vector2i{1, 0},    //
+    Eigen::Vector2i{1, 1},    //
 };
 
 /// Default neighbor kernel for the 3D NDT sensor model.
 const std::vector<Eigen::Vector3i> kDefaultNeighborKernel3d = {
     // TODO(Ramiro) revisit this kernel when we implement the 3D sensor model and extend this if it's computationally
     // feasible.
```
Comment (hidmic) [inline beluga/include/beluga/sensor/ndt_sensor_model.hpp:122]: @serraramiro1 :eyes:

File: beluga/test/beluga/sensor/test_ndt_model.cpp [hunk 2]
```diff
@@ -130,7 +132,85 @@ TEST(NDTSensorModelTests, ToCellsNotEnoughPointsInCell) {
...
+  ASSERT_EQ(cells.size(), 0UL);
+}
+
+TEST(NDTSensorModel3DTests, FitPoints) {
+  {
+    const std::vector meas{
+        Eigen::Vector3d{0.1, 0.2, 0.3}, Eigen::Vector3d{0.1, 0.2, 0.3}, Eigen::Vector3d{0.1, 0.2, 0.3},
+        Eigen::Vector3d{0.1, 0.2, 0.3}, Eigen::Vector3d{0.1, 0.2, 0.3}, Eigen::Vector3d{0.1, 0.2, 0.3},
+    };
+    auto cell = detail::fit_points<3>(meas);
```
Comment (hidmic) [inline beluga/test/beluga/sensor/test_ndt_model.cpp:152]: @serraramiro1 nit: doesn't template parameter deduction work here? Same for `to_cells()`.

File: beluga/test/beluga/sensor/test_ndt_model.cpp [hunk 3]
```diff
@@ -130,7 +132,85 @@ TEST(NDTSensorModelTests, ToCellsNotEnoughPointsInCell) {
...
+    ASSERT_TRUE(cell.mean.isApprox(Eigen::Vector3d{0.1, 0.34, 0.36}, 1e-6));
+    ASSERT_FALSE(cell.covariance.isZero());
+    ASSERT_GT(cell.covariance(1, 1), cell.covariance(0, 0));
+    ASSERT_GT(cell.covariance(1, 1), cell.covariance(2, 2));
+    ASSERT_GT(cell.covariance(2, 2), cell.covariance(0, 0));
+  }
+}
+
+TEST(NDTSensorModel3DTests, SensorModel) {
+  const double map_res = 0.5;
```
Comment (hidmic) [inline beluga/test/beluga/sensor/test_ndt_model.cpp:171]: @serraramiro1 nit: can be made `constexpr`.

File: beluga/include/beluga/sensor/ndt_sensor_model.hpp (hunk 1 above)
Comment (serraramiro1) [reply beluga/include/beluga/sensor/ndt_sensor_model.hpp:122]: Still unclear to me without testing this E2E on a node. :(

File: beluga/test/beluga/sensor/test_ndt_model.cpp [hunk 4]
```diff
@@ -112,31 +114,109 @@ TEST(NDTSensorModelTests, FitPoints) {
...
+    ASSERT_TRUE(cell.mean.isApprox(Eigen::Vector3d{0.1, 0.34, 0.36}, 1e-6));
+    ASSERT_FALSE(cell.covariance.isZero());
+    ASSERT_GT(cell.covariance(1, 1), cell.covariance(0, 0));
+    ASSERT_GT(cell.covariance(1, 1), cell.covariance(2, 2));
+    ASSERT_GT(cell.covariance(2, 2), cell.covariance(0, 0));
+  }
+}
+
+TEST(NDTSensorModel3DTests, SensorModel) {
+  constexpr double map_res = 0.5;
```
Comment (hidmic) [inline beluga/test/beluga/sensor/test_ndt_model.cpp:171]: @serraramiro1 nit: consider using Hungarian notation for consistency:
```suggestion
  constexpr double kMapResolution = 0.5;
```
Same elsewhere for other new `constexpr` in this file.

**Review by hidmic — APPROVED** [review]

File: beluga/test/beluga/sensor/test_ndt_model.cpp (hunk 4 above)
Comment (serraramiro1) [reply beluga/test/beluga/sensor/test_ndt_model.cpp]: DOne

**Review by hidmic — APPROVED** [review]
LGTM with green CI

**Pull request merged**